### DISCUSSION TITLE
How to acces widgets_map from a regular function?

### DISCUSSION THREAD

**Discussion by mcnuggetsx20** [qtile/qtile#3443] [Q&A] [answered]
I have a GenPollText widget that calls a function every second. I want this function to update a few widgets but apparently GenPollText does not pass the qtile object into the function, which prevents me from accessing widgets_map. Is there any other way to access the qtile object and the widgets_map?

**Comment by elParaguayo** [comment, marked as answer]
In your config, make sure you import `qtile` e.g.

``` python
from libqtile import qtile

def my_poll_func():
    widgets = qtile.widgets_map
    # etc. etc.
```
